windows/src/main/csharp/ch/cyberduck/ui/core/preferences/PackagedDataMigrator.cs

```diff
@@ -0,0 +1,102 @@
+
+        string path;
+        try
+        {
+            const KNOWN_FOLDER_FLAG FLAGS = KNOWN_FOLDER_FLAG.KF_FLAG_RETURN_FILTER_REDIRECTION_TARGET | KNOWN_FOLDER_FLAG.KF_FLAG_DONT_VERIFY;
+            path = SHGetKnownFolderPath(FOLDERID_RoamingAppData, FLAGS, default);
+        }
+        catch (Exception e)
+        {
+            logger.warn("Failure getting RoamingAppData packaged folder path", e);
+            return;
+        }
+
+        path = Path.Combine(path, Runtime.DataFolderName);
+        if (!Directory.Exists(path))
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Skipping migration for non-existent data folder");
+            }
+
+            return;
+        }
+
+        DirectoryInfo migratedLocation = new(Path.Combine(path, "Migrated"));
+        migratedLocation.Create();
+        var targetPath = RoamingSupportDirectoryFinder.Local.getAbsolute();
+        foreach (var item in new[] { "Bookmarks", "History", "Profiles", "Sessions", "Transfers" })
+        {
+            DirectoryInfo directory = new(Path.Combine(path, item));
+            if (!directory.Exists)
+            {
+                continue;
+            }
+
+            DirectoryInfo migratedDirectory = new(Path.Combine(migratedLocation.FullName, item));
+            migratedDirectory.Create();
+            var copyTarget = Path.Combine(targetPath, item);
+            foreach (var file in directory.EnumerateFiles())
+            {
+                bool failedMigrate = false;
+                FileInfo targetFile = new(Path.Combine(copyTarget, file.Name));
+                try
+                {
+                    if (targetFile.Exists && targetFile.LastWriteTime >= file.LastWriteTime)
+                    {
+                        // already migrated, or newer on shared location
+                        // move file to migrated-location
+
+                        if (logger.isInfoEnabled())
+                        {
+                            logger.info($"Skipping {item}/{file.Name} - Exists and target is newer");
+                        }
+
+                        continue;
+                    }
+
+                    try
+                    {
+                        file.CopyTo(targetFile.FullName, true);
+                    }
+                    catch (Exception e)
+                    {
+                        failedMigrate = true;
+                        logger.warn($"Failure migrating {item}/{file.Name}", e);
+                    }
+                }
+                finally
+                {
+                    if (!failedMigrate)
+                    {
+                        // skip moving when file couldn't be copied to shared lcoation
+                        try
+                        {
+                            // Move with overwrite
+                            file.CopyTo(Path.Combine(migratedDirectory.FullName, file.Name), true);
```

Comment:
Why not trash after migrating?

Reply:
Didn't want to delete files, that failed copying over to shared location - thus risking data loss.
Trashing might be an alternative.